packages/ember-glimmer/tests/integration/helpers/element-action-test.js

```diff
@@ -0,0 +1,1340 @@
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.equal(fooCallCount, 2, 'foo has been called 2 times');
+  }
+
+  ['@test it can call an action with parameters']() {
+    let fooArgs = [];
+    let component;
+
+    let ExampleComponent = Component.extend({
+      member: 'a',
+      init() {
+        this._super(...arguments);
+        component = this;
+      },
+      actions: {
+        foo(thing) {
+          fooArgs.push(thing);
+        }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<button {{action "foo" member}}>Click me</button>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.assert.deepEqual(fooArgs, [], 'foo has not been called');
+
+    this.runTask(() => this.rerender());
+
+    this.assert.deepEqual(fooArgs, [], 'foo has not been called');
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.deepEqual(fooArgs, ['a'], 'foo has not been called');
+
+    this.runTask(() => {
+      component.set('member', 'b');
+    });
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.deepEqual(fooArgs, ['a', 'b'], 'foo has been called with an updated value');
+  }
+
+  ['@htmlbars it should output a data attribute with a guid']() {
+    this.render('<button {{action "show"}}>me the money</button>');
+
+    let button = this.$('button');
+
+    this.assert.ok(button.attr('data-ember-action').match(/\d+/), 'A data-ember-action attribute with a guid was added');
+  }
+
+  ['@glimmer it should output a marker data attribute and a data attribute with a guid']() {
+    this.render('<button {{action "show"}}>me the money</button>');
+
+    let button = this.$('button');
+    let attributes = getActionAttributes(button.get(0));
+
+    this.assert.ok(button.attr('data-ember-action').match(''), 'An empty data-ember-action attribute was added');
+    this.assert.ok(attributes[0].match(/data-ember-action-\d+/), 'A data-ember-action-xyz attribute with a guid was added');
+  }
+
+  ['@skip it should by default register a click event']() {
+    // TODO
+    // Should we be testing implementation details? Why?
+  }
+
+  ['@skip it should allow alternative events to be handled']() {
+    // TODO
+    // Should we be testing implementation details? Why?
+  }
+
+  ['@skip it should by default target the view\'s controller']() {
+    // TODO
+    // Should we be testing implementation details? Why?
+  }
+
+  ['@skip inside a yield, the target points at the original target']() {
+    // TODO
+    // Testing with views and {{#view}}?
+  }
+
+  ['@test it should allow a target to be specified']() {
+    let targetWatted = false;
+
+    let TargetComponent = Component.extend({
+      actions: {
+        wat() {
+          targetWatted = true;
+        }
+      }
+    });
+
+    let target = TargetComponent.create();
```

Comment:
I would prefer to avoid manually calling `.create` on components.  One possible alternative to this would be:

```js
this.registerComponent('parent-target', {
  ComponentClass: TargetComponent,
  template: '{{yield this}}'
});

// ...snip...
this.render(`{{#parent-target as |parent|}}{{other-component anotherTarget=parent}}{{/parent-target}}`);
```